pythonwhat/State.py

```diff
@@ -395,5 +203,17 @@ def parse_int(x):
     def set_active_state(state):
         State.active_state = state
 
+# add property methods for retrieving parser outputs
+# e.g. state.student_withs
+from functools import partial
+for k, Parser in parser_dict.items():
+    def getx(self, tree_name, Parser=Parser): 
+        p = Parser()
+        p.visit(getattr(self, tree_name))
+        return p.out
+
+    for s in ['student', 'solution']:
+        setattr(State, s+'_'+k, property(partial(getx, tree_name = s+'_tree')))
```

Comment:
You are positive that this approach does not lead to additional re-parsing if you use `test_function()` twice in an SCT right? As soon as it's computed, it's not recomputed? Can you maybe add some comments to the magic that's happening here? It's a bit obscure now.